### ISSUE TITLE
[BUG] Can't open camera on adding new element

### ISSUE THREAD

**Issue by Linskun** [davidhealey/waistline#733]
Hello,

i have a Pixel 6a with latest Android 13 update.
I can't add a picture on adding new element/food, the UI say that "There was a problem using the camera"
If i use camera to scan a bar code it work, so It doesn't look like a permissions issue

**Comment by Linskun** [comment]
No, my phone is stock with latest Google official firmware and official photo apps

**Comment by licaon-kter** [comment]
Seeing the same on Divest OS 20 (aka Android 13) based on Lineage

Trying the older version, qhich opens the camera fine, it appears that **now** you don't ask for Storage permissions of any kind (Photos and whatnot)?

Testing on Android 11 I get asked for Storage on start and the camera works fine.

**Comment by davidhealey** [comment]
I'm using Android 13 (Lineage) on my Pixel 6a without issue. So I don't think it's a universal problem.

**Comment by licaon-kter** [comment]
@davidhealey
> So I don't think it's a universal problem.

So you've installed fresh today and works fine for you?

**Comment by davidhealey** [comment]
I haven't reinstalled it today, I'm using the latest version from fdroid.

**Comment by davidhealey** [comment]
Aha ok I'll try that. @EmilJunker I seem to recall you updated the SDK version?

**Comment by Linskun** [comment]
I've tried also the latest downloaded from GitHub

Did you see my video ?
The App have already authorization on camera (Default official Google gcam)
If i click on barcode It work
If i try to add a new food elementi It don't work

Tell me if you Need other information, logcat

**Comment by licaon-kter** [comment]
previous:  
![previous](https://github.com/davidhealey/waistline/assets/2282799/9a0cdf20-8f6b-4edf-b75a-d20bced5b7fe)


vs.

now:
![now](https://github.com/davidhealey/waistline/assets/2282799/c4594346-cd6d-4b81-9e32-7583ce5e1daf)

**Comment by Linskun** [comment]
Now on official Google Firmware
![Screenshot_20230807-131118](https://github.com/davidhealey/waistline/assets/45461032/64bd1e4b-4839-4ae3-9581-266d3a5452a8)

**Comment by EmilJunker** [comment]
@Linskun @licaon-kter Are you able to create a database backup from `Settings > Import/Export > Backup Database`? That should also require storage access permission.

**Comment by Linskun** [comment]
@EmilJunker : Export Failed
And i think that Is normal, the App don't ask storage permissions

**Comment by EmilJunker** [comment]
Or actually we may have to wait for the camera plugin itself to get fixed, see https://github.com/apache/cordova-plugin-camera/issues/825#issuecomment-1613588878

**Comment by EmilJunker** [comment]
The main reason was because I wanted to add support for themed app icons (issue #675) which is a new feature in Android 13 that requires Cordova Android 12 and SDK 33.

**Comment by davidhealey** [comment]
Ah yes, I remember now, thanks

**Comment by EmilJunker** [comment]
Unfortunately yes, from August 31, apps submitted to the Play Store are required to target at least SDK 33: https://developer.android.com/google/play/requirements/target-sdk

On the plus side, that would give us three more weeks to make and publish a version that targets SDK 32 (which is unaffected by the issue).

To make a working version that targets SDK 33, it looks like we'll need to wait for the camera plugin and the file plugin to be fixed:

- https://github.com/apache/cordova-plugin-camera/issues/825
- https://github.com/apache/cordova-plugin-file/issues/586

**Comment by davidhealey** [comment]
Uh why they always breaking stuff!!! :)

**Comment by licaon-kter** [comment]
Wait, ain't `Android/data` unaccessible by other apps? How can I _backup the backup files_ ?!

**Comment by EmilJunker** [comment]
@licaon-kter Yes, some file managers block access to `Android/data`. But there are also file manager apps that let you access the folder. So if you don't see it in the default files app on your phone, try a different one.

**Comment by licaon-kter** [comment]
You got it backwards, Google blocks their access... but yeah.

Are you really really sure Documents is blocked?

**Comment by EmilJunker** [comment]
> Are you really really sure Documents is blocked?

I couldn't find a way to write to the Documents folder on Android 13. Maybe there is a way to do it with a native app, but with Cordova we have to rely on plugins to provide this functionality. You are welcome to try to make it work, though. The app is open source ;)

**Comment by licaon-kter** [comment]
>  Maybe there is a way to do it with a native app, but with Cordova we have to rely on plugins to provide this functionality.

Ah, so a matter of time :)

**Comment by davidhealey** [comment]
Hi,

I've incorporated Emil's temporary fix, but I since I had to install some new stuff I would appreciate if a few of you could test it before I make an official release.

[waistline-release.zip](https://github.com/davidhealey/waistline/files/12337245/waistline-release.zip)

**Comment by EmilJunker** [comment]
@davidhealey I installed your APK on my S21 running Android 13 and can confirm that it's fixed. I can add pictures to food items and the backup is also working as expected.

**Comment by davidhealey** [comment]
Perfect thanks, in that case I'll go ahead and make the release.

**Comment by schklom** [comment]
I have no experience with coding for Android, but is there a chance the fix described at https://stackoverflow.com/questions/38832592/cordova-file-plugin-save-file-in-device/38900101#38900101 can work here?
